python/MDSplus/tree.py

```diff
@@ -1711,6 +1711,62 @@ def beginSegment(self, start, end, dim, array, idx=-1):
                                        _dat.Data.byref(array),
                                        _C.c_int32(int(idx))))
 
+    def beginSegmentResampled(self,start,end,dim,array,resNode,resFactor,idx=-1):
+        """Begin a resampled record segment, using an average, rows_filled = 0
+        @param start: Index of first row of data
+        @type start: Data
+        @param end: Index of last row of data
+        @type end: Data
+        @param dim: Dimension information of segment
+        @type dim: Dimension
+        @param array: Initial data array. Defines shape of segment
+        @type array: Array
+        @param resNode: TreeNode of the node to write the resampled data into
+        @type resNode: TreeNode
+        @param resFactor: Number of samples to resample together
+        @type resFactor: Data
+        @rtype: None
+        """
+        start,end,dim,array = map(_dat.Data,(start,end,dim,array))
+        _exc.checkStatus(
+            _TreeShr._TreeBeginSegmentResampled(self.ctx,
+                                                self._nid,
+                                                _dat.Data.byref(start),
+                                                _dat.Data.byref(end),
+                                                _dat.Data.byref(dim),
+                                                _dat.Data.byref(array),
+                                                _C.c_int32(int(idx)),
+                                                _C.c_int32(int(resNode._nid)),
+                                                _C.c_int32(int(resFactor))))
+
+    def beginSegmentMinMax(self,start,end,dim,array,resNode,resFactor,idx=-1):
+        """Begin a resampled record segment, using the min and max values
+        @param start: Index of first row of data
+        @type start: Data
+        @param end: Index of last row of data
+        @type end: Data
+        @param dim: Dimension information of segment
+        @type dim: Dimension
+        @param array: Initial data array. Defines shape of segment
+        @type array: Array
+        @param resNode: TreeNode of the node to write the resampled data into
+        @type resNode: TreeNode
+        @param resFactor: Number of samples to resample together
+        @type resFactor: Data
+        @rtype: None
+        """
+        start,end,dim,array = map(_dat.Data,(start,end,dim,array))
+        _exc.checkStatus(
+            _TreeShr._TreeBeginSegmentMinMax(self.ctx,
+                                             self._nid,
+                                             _dat.Data.byref(start),
+                                             _dat.Data.byref(end),
+                                             _dat.Data.byref(dim),
+                                             _dat.Data.byref(array),
+                                             _C.c_int32(int(idx)),
+                                             _C.c_int32(int(resNode._nid)),
```

Comment:
_nid is a ctypes pointer kind under python3 this wont work.. passing it directly like the second arguments should work though
```python
resNode._nid,
```

Reply:
could not get it to pass teh test though
```python
======================================================================
ERROR: runTest (MDSplus.tests.segmentsUnitTest.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/c/git/mdsplus/python/MDSplus/tests/_UnitTest.py", line 77, in runTest
    self.__getattribute__(test)()
  File "/c/git/mdsplus/python/MDSplus/tests/segmentsUnitTest.py", line 141, in WriteSegments
    self.assertEqual(resNode.getSegmentLimits(1),(3,5))
  File "/c/git/mdsplus/python/MDSplus/tree.py", line 2269, in getSegmentLimits
    end.ref))
  File "/c/git/mdsplus/python/MDSplus/mdsExceptions.py", line 94, in checkStatus
    raise exception
TreeNOSEGMENTS: %TREE-E-NOSEGMENTS, No segments exist in this node

----------------------------------------------------------------------
Ran 1 test in 1.146s

FAILED (errors=1)
FAIL segmentsUnitTest.py (exit status: 1)
```